zrml/neo-swaps/src/math.rs

```diff
@@ -606,4 +712,36 @@ mod tests {
             exp(FixedType::checked_from_num(EXP_OVERFLOW_THRESHOLD).unwrap(), neg).unwrap();
         assert_eq!(result, expected);
     }
+
+    #[test_case(_9_10, _10, _1_10, 219722457734)] // Large price shift
+    #[test_case(_4_10, _10, _3_10, 15415067983)] // Small price shift
+    #[test_case(_3_10, _10, _4_10, 0)] // Zero buy amount
+    #[test_case(_4_10, _10, _4_10, 0)] // Zero buy amount
```

Comment:
We should extend these tests by the following four cases: Very large/small `until`, very large/small `spot_price`.

Reply:
9be4ccbe905b134cc7e3c4bd749c0c7520f4aaa7